flatland_plugins/test/dynamics_limits_test.cpp

```diff
@@ -0,0 +1,316 @@
+  EXPECT_NEAR(0.45, result, 1e-3);
+  result = dynamics_limits.Limit(1.0, 0.0, 0.05);  // Try to deccelerate at -20m/s/s, when the limit is 9
+  EXPECT_NEAR(0.55, result, 1e-3);
+  result = dynamics_limits.Limit(10.0, 0.0, 0.05);  // Try to deccelerate at -20m/s/s, when the limit is 9
+  EXPECT_NEAR(9.55, result, 1e-3);
+  result = dynamics_limits.Limit(-10.0, 0.0, 0.05);  // Try to deccelerate at 20m/s/s, when the limit is 9
+  EXPECT_NEAR(-9.55, result, 1e-3);
+  result = dynamics_limits.Limit(5, -5, 0.05);  // Try to deccelerate at 20m/s/s, when the limit is 9
+  EXPECT_NEAR(4.55, result, 1e-3);
+  result = dynamics_limits.Limit(-3, -2, 0.05);  // Try to deccelerate at 20m/s/s, when the limit is 9
+  EXPECT_NEAR(-2.55, result, 1e-3);
+}
+
+TEST(DynamicsLimitsTest, test_Saturate) {
+  auto dynamics_limits = DynamicsLimits();
+  double result;
+  // Useless, strict bounds
+  result = dynamics_limits.Saturate(0, 0, 0);  // Bound between 0 and 0, 0
+  EXPECT_NEAR(0, result, 1e-3);
+  result = dynamics_limits.Saturate(1, 0, 0);  // Bound between 1 and 0, 0
+  EXPECT_NEAR(0, result, 1e-3);
+  result = dynamics_limits.Saturate(-1, 0, 0);  // Bound between -1 and 0, 0
+  EXPECT_NEAR(0, result, 1e-3);
+
+  // Valid bounds
+  result = dynamics_limits.Saturate(0, -1.1, 2.2);
+  EXPECT_NEAR(0, result, 1e-3);
+  result = dynamics_limits.Saturate(1, -1.1, 2.2);
+  EXPECT_NEAR(1, result, 1e-3);
+  result = dynamics_limits.Saturate(-1,  -1.1, 2.2);
+  EXPECT_NEAR(-1, result, 1e-3);
+  result = dynamics_limits.Saturate(3, -1.1, 2.2);
+  EXPECT_NEAR(2.2, result, 1e-3);
+  result = dynamics_limits.Saturate(-3,  -1.1, 2.2);
+  EXPECT_NEAR(-1.1, result, 1e-3);
+
+  // Invalid bounds, input not changed
+  result = dynamics_limits.Saturate(0, 1.1, -2.2);
+  EXPECT_NEAR(0, result, 1e-3);
+  result = dynamics_limits.Saturate(1, 1.1, -2.2);
+  EXPECT_NEAR(1, result, 1e-3);
+  result = dynamics_limits.Saturate(-1,  1.1, -2.2);
+  EXPECT_NEAR(-1, result, 1e-3);
+  result = dynamics_limits.Saturate(3, 1.1, -2.2);
+  EXPECT_NEAR(3, result, 1e-3);
+  result = dynamics_limits.Saturate(-3,  1.1, -2.2);
+  EXPECT_NEAR(-3, result, 1e-3);
+}
+
+/**
+ * Test Limit function wrt. acceleration asnd decelleration
+ */
+TEST(DynamicsLimitsTest, test_Limit_deceleration) {
+  auto dynamics_limits = DynamicsLimits();
+
+  YAML::Node config = YAML::Node();
+  config["acceleration_limit"] = 1.0;
+  config["deceleration_limit"] = 9.0;
+  dynamics_limits.Configure(config);
+
+  EXPECT_NEAR(1.0, dynamics_limits.acceleration_limit_, 1e-3);
+  EXPECT_NEAR(9.0, dynamics_limits.deceleration_limit_, 1e-3);
+  EXPECT_NEAR(0.0, dynamics_limits.velocity_limit_, 1e-3);
+
+  double result;
+  result = dynamics_limits.Limit(0, 1.0, 0.05);  // Try to accelerate at 20m/s/s, when the limit is 1 (acceleration)
+  EXPECT_NEAR(0.05, result, 1e-3);
+  result = dynamics_limits.Limit(0.97, 1.0, 0.05);  // Try to accelerate at <1m/s/s, when the limit is 1
```

Comment:
What is the < sign for in the comment?

Reply:
Fixed, not sure what I intended the < to mean here, so I removed it